modules/dxp/apps/search-experiences/search-experiences-web/src/main/resources/META-INF/resources/sxp_blueprint_admin/js/shared/sxp_element/category_selector_input/index.js

```diff
@@ -98,6 +117,12 @@ function CategorySelectorInput({
 	const [autocompleteDropdownActive, setAutocompleteDropdownActive] =
 		useState(false);
 
+	const _getIdentifierType = () =>
+		id.includes(ASSET_CATEGORY_EXTERNAL_REFERENCE_CODE) ||
+		id.includes(ASSET_CATEGORY_EXTERNAL_REFERENCE_CODES)
+			? IDENTIFIER_TYPES.EXTERNAL_REFERENCE_CODE
+			: IDENTIFIER_TYPES.ID;
```

Comment:
Looking at this again, it might be a good idea to wrap this in a `useMemo` since`_getIdentifierType()` being used in a loop.

We can easily wrap this and the everywhere that calls `_getIdentifierType()` will only need to return the cached value.
```
const _getIdentifierType = useMemo(() => {
  return (
    id.includes(ASSET_CATEGORY_EXTERNAL_REFERENCE_CODE) ||
    id.includes(ASSET_CATEGORY_EXTERNAL_REFERENCE_CODES)
  )
    ? IDENTIFIER_TYPES.EXTERNAL_REFERENCE_CODE
    : IDENTIFIER_TYPES.ID;
}, [id]);
```

Reply:
Added the memoizing! Let me know if there's any issues? I applied the same idea to `SiteSelectorInput` as well, to be consistent. 

I guess with `useMemo` it is gets to be a value, so `_getIdentifierType` renamed to `identifierType` too 😅

Reply:
Looks good! 💯